main.tf

```
@@ -30,21 +30,28 @@ locals {
   })
 }
 
-resource "local_file" "vcsa_template_to_json" {
-  filename = "/tmp/vcsa/vcsa-${var.deploy_type}.json"
-  content  = local.vcsa_template
-}
+# resource "local_sensitive_file" "vcsa_template_to_json" {
+#   filename       = "/tmp/vcsa/vcsa-${var.deploy_type}.json"
+#   content        = local.vcsa_template
+# }
 
 resource "null_resource" "vcsa_linux_deploy" {
-  count = var.windows == false ? 1 : 0
+  count = var.linux == true ? 1 : 0
   provisioner "local-exec" {
-    command = "${local.binaries_path}/vcsa-cli-installer/lin64/vcsa-deploy install --accept-eula --acknowledge-ceip --no-ssl-certificate-verification /tmp/vcsa/vcsa-${var.deploy_type}.json"
+    command = "${local.binaries_path}/vcsa-cli-installer/lin64/vcsa-deploy install --accept-eula --acknowledge-ceip --no-ssl-certificate-verification ${local.vcsa_template}"
   }
 }
 
 resource "null_resource" "vcsa_windows_deploy" {
   count = var.windows == true ? 1 : 0
   provisioner "local-exec" {
-    command = "${local.binaries_path}/vcsa-cli-installer/win32/vcsa-deploy.exe install --accept-eula --acknowledge-ceip --no-ssl-certificate-verification /tmp/vcsa/vcsa-${var.deploy_type}.json"
+    command = "${local.binaries_path}/vcsa-cli-installer/win32/vcsa-deploy.exe install --accept-eula --acknowledge-ceip --no-ssl-certificate-verification ${local.vcsa_template}"
+  }
+}
+
+resource "null_resource" "vcsa_mac_deploy" {
+  count = var.mac == true ? 1 : 0
```

```suggestion
  count = var.mac ? 1 : 0
```

Same comment as the previous one for `linux`.